Qt also bundles libz

Example issue when building sme:

```
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/smelibs/lib/libQt6BundledZLIB.a(crc32.c.obj):crc32.c:(.text+0x560): multiple definition of `crc32_combine_gen64'; C:/smelibs/lib/libz.a(crc32.obj):crc32.c:(.text+0xc00): first defined here
```

Not sure why this error didn't occur previously - but if possible we should remove our libz and link to the Qt one wherever libz is needed.

Should be straightforward - we currently don't use any cmake config for zlib but just pass in location of our statically compiled lib and headers, and Qt uses the same location and names for all operating systems:

- `lib/libQt6BundledZLIB.a`
- `include/QtZlib/zlib.h`

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -4,7 +4,7 @@ on: push
 
 env:
   LLVM_VERSION: "14.0.6"
-  QT_VERSION: "6.3.1"
+  QT_VERSION: "6.3.1-zlib"
   LIBSBML_VERSION: "development"
   LIBEXPAT_VERSION: "R_2_4_8"
   SYMENGINE_VERSION: "v0.9.0"
@@ -23,7 +23,6 @@ env:
   BOOST_VERSION_: "1_79_0"
   QCUSTOMPLOT_VERSION: "2.1.0fixed"
   CEREAL_VERSION: "v1.3.2"
-  ZLIB_VERSION: "v1.2.12"
   PAGMO_VERSION: "v2.18.0"
   BZIP2_VERSION: "1.0.8"
 

--- a/README.md
+++ b/README.md
@@ -21,7 +21,6 @@ This repo provides the following statically compiled libraries:
 - [Boost](https://www.boost.org/)
 - [QCustomPlot](https://www.qcustomplot.com)
 - [Cereal](https://github.com/USCiLab/cereal)
-- [zlib](https://zlib.net/)
 - [bzip2](https://www.sourceware.org/bzip2/)
 - [pagmo](https://github.com/esa/pagmo2)
 - [LLVM](https://llvm.org/) (copied from <https://github.com/spatial-model-editor/sme_deps_llvm>)

--- a/build.sh
+++ b/build.sh
@@ -24,7 +24,8 @@ echo "BOOST_VERSION: ${BOOST_VERSION}"
 echo "BOOST_VERSION_: ${BOOST_VERSION_}"
 echo "QCUSTOMPLOT_VERSION: ${QCUSTOMPLOT_VERSION}"
 echo "CEREAL_VERSION: ${CEREAL_VERSION}"
-echo "ZLIB_VERSION: ${ZLIB_VERSION}"
+echo "PAGMO_VERSION: ${PAGMO_VERSION}"
+echo "BZIP2_VERSION: ${BZIP2_VERSION}"
 
 NPROCS=2
 echo "NPROCS: ${NPROCS}"
@@ -66,26 +67,6 @@ cd bzip2-${BZIP2_VERSION}
 make CFLAGS="-O2 -g -D_FILE_OFFSET_BITS=64 -fPIC" -j$NPROCS
 make install PREFIX="$INSTALL_PREFIX"
 cd ../
-
-# build static version of zlib
-git clone -b $ZLIB_VERSION --depth 1 https://github.com/madler/zlib.git
-cd zlib
-mkdir build
-cd build
-cmake -G "Unix Makefiles" .. \
-    -DCMAKE_OSX_DEPLOYMENT_TARGET="10.14" \
-    -DCMAKE_BUILD_TYPE=Release \
-    -DBUILD_SHARED_LIBS=OFF \
-    -DCMAKE_C_FLAGS="-fpic -fvisibility=hidden" \
-    -DCMAKE_CXX_FLAGS="-fpic -fvisibility=hidden" \
-    -DCMAKE_INSTALL_PREFIX="$INSTALL_PREFIX"
-time make zlibstatic -j$NPROCS
-# manual install to avoid shared libs being installed & issues with compiling example programs
-# wildcard is used because on mingw it calls the library file libzlibstatic.a for some reason:
-cp libz*.a $INSTALL_PREFIX/lib/libz.a
-cp zconf.h $INSTALL_PREFIX/include/.
-cp ../zlib.h $INSTALL_PREFIX/include/.
-cd ../../
 
 # install Cereal headers
 git clone -b $CEREAL_VERSION --depth 1 https://github.com/USCiLab/cereal.git

